fixed type error on breakdown filter

EDsCODE opened the pull request:
Addresses #759 
 
## Changes
- fixed numeric breakdown values not working

## Checklist
- [ ] All querysets/queries filter by Team (if applicable)
- [ ] Backend tests (if applicable)

EDsCODE committed:
fixed type error

EDsCODE committed:
drop non values

EDsCODE committed:
don't drop

EDsCODE committed:
change label to friendly

EDsCODE commented:
@timgl This should solve the issue but I found something funny that we should address. When you breakdown by a property, are the leftovers (anything that doesn't have the property) supposed to be displayed? 

Right now, if you breakdown by browser_version it will give you the line with the browser version but also a line with 'nan'. Also, when clicking on the points of the 'nan' line even if the datapoint is 0, a list of users will be query-able.

EDsCODE committed:
update tests

timgl commented:
i don't have a strong opinion either way. I think it can be useful to have the non-set line too so if it's easy to make it work.

Re the 0 being clickable, it just shows '0 users found' so I think that's fine. I did find a bug where setting properties on entities doesn't correctly filter /people, will create a ticket for that

EDsCODE committed:
work for numerical

EDsCODE commented:
Fixed! The 0 person error was happening on numerical breakdowns also. So I was seeing a line with 0 but when you clicked on a datapoint you'd see actual people related to the event. The problem was that when something was nan we were tossing the results rather than aggregating them. I changed it so that it treats nan and nones as strings and become a breakdown category which will be displayed as "other" in the label

timgl commented on posthog/api/action.py:
```diff
@@ -330,10 +331,10 @@ def _stickiness(self, filtered_events: QuerySet, entity: Entity, filter: Filter)
         }
 
-    def _breakdown_label(self, entity: Entity, value: str) -> Dict[str, Optional[str]]:
+    def _breakdown_label(self, entity: Entity, value: Any) -> Dict[str, Optional[str]]:
```
Any tends to be an antipattern. Is it actually `Union[str, int]`?

EDsCODE replied:
yup, forgot about Union. thanks

EDsCODE committed:
change any to union

EDsCODE committed:
correct all types
